v0.23.0: cherry-pick docs updates from master (#2210)
firecracker-microvm/firecracker: docs/design.md

(MMDS) to the guest through the API endpoint. The metadata stored by the
service is fully configured by users.

### Sandboxing

#### __Firecracker process__

##### Seccomp

Seccomp filters are used by default to further limit the system calls Firecracker
can use. There are 3 possible levels of seccomp filtering, configurable by passing
a command line argument to Firecracker: 0 (disabled), 1 (whitelists a set of
trusted system calls by their identifiers) and 2 (whitelists a set of trusted
system calls with trusted parameter values), the latter being the most

> restrictive and the recommended one. The filters are loaded in the Firecracker
> process, immediately before the execution of the untrusted guest code starts.

Reviewer: This is a bit confusing, do we really trust any guest code ?
```suggestion
recommended and restricted one. Firecracker enables the filters on all threads 
before the execution of guest code starts.
```

Author: We don't trust any guest code. We just emphasise the fact that the guest code is untrusted.

Author: the phrasing above is actually the exact same phrasing used on master, even before the docs update commit. you think it makes sense to alter it here? My opinion is that we should open a PR on mainline about any issues regarding this phrasing

Reviewer: I think we can downgrade this as a nit.
